galaxyproject/galaxy
# Expose more tool information / navigability in UI.

We had a specific request that job parameter displays that:

> For: Job parameters as table. The output doesn't show what tool was used, it would be good if it did.

I don't want to populate the widget directly with information because it might disrupt the flow of the report but I agree it should be navigable. (See https://github.com/galaxyproject/galaxy/pull/16994 for a discussion of why I think making report components minimally visibly disruptive is important.)

So I have augmented the widget such that is a title is added - a little info icon will appear in the title span that will show a link to the tool and render the name as it would appear in the tool form. This information should be available in other places for the same reason IMO. So this popover is now used in the job metrics table the same way as well as in the workflow step display Markdown component and in the invocation step display component.

Used in the invocation display:

<img width="461" alt="Screenshot 2023-11-29 at 12 52 01 PM" src="https://github.com/galaxyproject/galaxy/assets/216771/4550c994-ae42-41f5-8360-99be794caf49">

Used in the workflow display Markdown component

<img width="410" alt="Screenshot 2023-11-29 at 12 51 42 PM" src="https://github.com/galaxyproject/galaxy/assets/216771/741e4ac9-3eea-4816-be61-3a64341d6314">

Used in the job parameters display component

<img width="459" alt="Screenshot 2023-11-29 at 12 51 35 PM" src="https://github.com/galaxyproject/galaxy/assets/216771/9f483b8e-1f8e-45cd-b806-37ab1cf19c91">

This PR builds on the refactoring outlined in #17004 to allow the reuse across these components as well a commit that adds title and footer options to job parameters and job metrics components and #16681 to enable those.

## How to test the changes?
(Select all options that apply)
  1. Run a workflow with a tool step and see the link.

## License
